### ISSUE TITLE
dial tcp 0.0.0.0:8080: connect: connection refused

### ISSUE THREAD

**Issue by msetodev** [caddyserver/caddy#5300]
I've got vaultwarden and caddy running as separate docker containers

How I started vaultwarden:

sudo docker run -d --name vaultwarden -v /srv/vaultwarden:/data -e WEBSOCKET_ENABLED=true -p 127.0.0.1:8080:80 -p 127.0.0.1:3012:3012 --restart on-failure vaultwarden/server:latest

My Caddyfile:

mydomainwhichisresolvingcorrectly.com {
encode gzip

reverse_proxy /notifications/hub/negotiate 0.0.0.0:8080
reverse_proxy /notifications/hub 0.0.0.0:3012
reverse_proxy 0.0.0.0:8080}
}

How I started caddy:

sudo mkdir /etc/caddy

sudo chmod go-rwx /etc/caddy

sudo docker run -d -p 80:80 -p 443:443 --name caddy -v /etc/Caddyfile:/etc/caddy/Caddyfile -v /etc/caddy:/root/.local/share/caddy -v caddy_data:/data --restart on-failure caddy:2

Every time I try to access the domain via HTTPS, I see this in the caddy logs:

`{"level":"error","ts":1673468600.1370056,"logger":"http.log.error","msg":"dial tcp 0.0.0.0:8080: connect: connection refused"`

**Comment by mholt** [comment]
Thanks for your question, and we're glad that you're using Caddy! So far, this doesn't look like a bug in Caddy. It looks more like a question about how to use Caddy rather than a bug report or feature request.

Since this issue tracker is reserved for actionable development items, I'm going to close this, but we have a [community forum at caddy.community](https://caddy.community) where more people will be exposed to your question, including people who may be more expert or experienced than I am with the specific issue you're facing. I hope you'll ask your question there. The Help category in the forum even provides a template you can fill out.

If I'm wrong, and this does turn out to be a bug after further investigation, we can reopen this. Thanks for understanding!

**Comment by francislavoie** [comment]
"localhost" (or 127.0.0.1, or 0.0.0.0) inside of a Docker container means "this container". You need to use Docker networking to communicate to the other container. You can use the container name (via Docker's built-in DNS resolver) as long as both containers are in the same Docker network.

**Comment by webgtx** [comment]
tl;dr :shipit: 
use `host` in network_mode

> like this 

```bash
docker run -itd --rm --name caddyserver --network=host caddy
```

**Comment by francislavoie** [comment]
@webgtx that's not necessarily the right solution. If you need to connect to another container, use that container's name. If you need to connect to the host, use `host.docker.internal` (google it).

**Comment by webgtx** [comment]
> @webgtx that's not necessarily the right solution. If you need to connect to another container, use that container's name. If you need to connect to the host, use `host.docker.internal` (google it).

Why just don't use host network mode? What could possibly go wrong with that ?

**Comment by francislavoie** [comment]
Using `host` mode throws away the point of using Docker's networking, i.e. isolation. It should be a last resort.
